dipy/workflows/reconst.py

```diff
@@ -0,0 +1,151 @@
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bvalues : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        bvectors : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        mask_files : string
+            Path to the input masks. This path may contain wildcards to use
+            multiple masks at once. (default: No mask used)
+        b0_threshold : float, optional
+            Threshold used to find b=0 directions (default 0.0)
+        out_dir : string, optional
+            Output directory (default input file directory)
+        out_tensor : string, optional
+            Name of the tensors volume to be saved (default 'tensors.nii.gz')
+        out_fa : string, optional
+            Name of the fractionnal anisotropy volume to be saved
+            (default 'fa.nii.gz')
+        out_ga : string, optional
+            Name of the geodesic anisotropy volume to be saved
+            (default 'ga.nii.gz')
+        out_rgb : string, optional
+            Name of the color fa volume to be saved (default 'rgb.nii.gz')
+        out_md : string, optional
+            Name of the mean diffusivity volume to be saved
+            (default 'md.nii.gz')
+        out_ad : string, optional
+            Name of the axial diffusivity volume to be saved
+            (default 'ad.nii.gz')
+        out_rd : string, optional
+            Name of the radial diffusivity volume to be saved
+            (default 'rd.nii.gz')
+        out_mode : string, optional
+            Name of the mode volume to be saved (default 'mode.nii.gz')
+        out_evec : string, optional
+            Name of the eigen vectors volume to be saved
+            (default 'evecs.nii.gz')
+        out_eval : string, optional
+            Name of the eigen vvalues to be saved (default 'evals.nii.gz')
+        """
+
+        io_it = self.get_io_iterator()
+
+        for dwi, bval, bvec, mask, otensor, ofa, oga, orgb, omd, oad, ord, \
+            omode, oevecs, oevals in io_it:
+
+            logging.info('Computing dti metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.get_affine()
+
+            if mask is None:
+                mask = None
+            else:
+                mask = nib.load(mask).get_data().astype(np.bool)
+
+            tenfit, _ = get_fitted_tensor(data, mask, bval, bvec, b0_threshold)
+
+            FA = fractional_anisotropy(tenfit.evals)
+            FA[np.isnan(FA)] = 0
+            FA = np.clip(FA, 0, 1)
+
+            tensor_vals = lower_triangular(tenfit.quadratic_form)
+            correct_order = [0, 1, 3, 2, 4, 5]
```

Comment:
Yeah - that's probably where it belongs, if it's useful in other places. 

I see now that this not in dti.py. Instead there's these other reordering variables: 

https://github.com/nipy/dipy/blob/master/dipy/reconst/dti.py#L1734
https://github.com/nipy/dipy/blob/master/dipy/reconst/dti.py#L1759

But not this one.

BTW - why is this the "correct order"?